### ISSUE TITLE
There may be bug in area_score function for the gtp command final_score

### ISSUE THREAD

**Issue by Splee99** [gcp/leela-zero#624]
As pointed out in the reddit message https://www.reddit.com/r/cbaduk/comments/7p6mlb/interesting_tromptaylor_fail/, sometimes the self play or match game results were wrong after both black and white passed. After replaying the games I found that both black and white generated their move normally, but after both sides played PASS, the gtp command final_score calls a separate function in FastBoard.cpp to calculate score:

// Needed for scoring passed out games not in MC playouts
float FastBoard::area_score(float komi) {
    auto white = calc_reach_color(WHITE);
    auto black = calc_reach_color(BLACK);

    auto score = -komi;

    for (int i = 0; i < m_boardsize; i++) {
        for (int j = 0; j < m_boardsize; j++) {
            auto vertex = get_vertex(i, j);

            if (white[vertex] && !black[vertex]) {
                score -= 1.0f;
            } else if (black[vertex] && !white[vertex]) {
                score += 1.0f;
            }
        }
    }

    return score;
}

I guess there may be some bug here resulting the wrong score mentioned in the reddit message?

**Comment by grolich** [comment]
You seem to have misunderstood that Reddit thread. There is no problem with the score after both passes. 
Tromp Taylor scoring should indeed deny white all points where white fails to remove the black stone. The given score is correct.

The author of the thread was simply puzzled over why, even if the NN still hasn't learned to not pass there as white because of the ruleset used, did the Search not find it despite it being so close.

While it is indeed interesting that such a big loophole exist in some positions in leela, it's not entirely unreasonable to expect a few. 

Bottom line: 
Whether this is the result of a blind spot of the NN, or a bug in the MCTS, the score given was the correct one under tromp Taylor rules

**Comment by Splee99** [comment]
Could you please clarify the function of area_score? It counts not only the stones, but also "enclosed" empty intersections as black or white. However, the "common" intersections are not counted. Is this also part of Tromp Taylor rule? By the way, this function is called when both side play PASS.

**Comment by Splee99** [comment]
Never mind, it is indeed how Tromp Taylor rule works.

**Comment by Splee99** [comment]
I replayed two of the games mentioned in Sabaki with LZ and the losing side resigned immediately. I guess in those games resign option was disabled. On the other hand, I would expect that the move PASS could have similar winrate as other moves in this situation because the game result is pretty fixed. In a normal go game, it doesn't make sense and sometime is seen as insulting the opponent to continue play. So playing PASS is normal to me. What puzzled me is that both black and white knew the game result, but the gtp command final_score generated debatable result.
